### PR TITLE
support excluding modules in gradualizer:type_check_file

### PR DESCRIPTION
**Pull request by tsloughter**
Not sure this is the best place to do it but I think it is the only place to do the filtering that doesn't require doing it in multiple places.

### CODE REVIEW

File: src/gradualizer.erl [hunk 1]
```diff
@@ -58,24 +58,31 @@ type_check_file(File) ->
...
-            case gradualizer_file_utils:get_forms_from_beam(File) of
-                {ok, Forms} ->
-                    type_check_forms(File, Forms, Opts);
-                Error ->
-                    throw(Error)
-            end;
-        Ext ->
-            throw({unknown_file_extension, Ext})
+    ExcludeModules = proplists:get_value(exclude_modules, Opts, []),
+    AtomBaseFile = list_to_atom(filename:rootname(filename:basename(File))),
```
Comment (zuiderkwast) [inline src/gradualizer.erl:62]: Perhaps we can avoid creating the atom until we know the extension is ".beam" or ".erl"?

**Review by zuiderkwast — COMMENTED** [review]
Welcome new contributor Tristan!

I think it looks like the right place to do it.

Next steps:

* Make the rebar3 plugin use this instead of its own logic.
* Docs (edoc somewhere, not dure where we list options)
* Add corresponding CLI flags

File: src/gradualizer.erl (hunk 1 above)
Comment (tsloughter) [reply src/gradualizer.erl:62]: Is that a common case? That'd only happen if a user passed in a filename to something that isn't a beam or erl file and not a big deal in that case to create the atom before it throws the exception.

Comment (zuiderkwast) [reply src/gradualizer.erl:62]: You're right, sorry.

**Review by erszcz — APPROVED** [review]
Looks good! This seems to be the best place to do the filtering.

**Pull request merged**